client/ayon_core/hosts/maya/api/yeti.py

```diff
@@ -0,0 +1,101 @@
+from typing import List
+
+from maya import cmds
+
+
+def get_yeti_user_variables(yeti_shape_node: str) -> List[str]:
+    """Get user defined yeti user variables for a `pgYetiMaya` shape node.
+
+    Arguments:
+        yeti_shape_node (str): The `pgYetiMaya` shape node.
+
+    Returns:
+        list: Attribute names (for a vector attribute it only lists the top
+            parent attribute, not the attribute per axis)
+    """
+
+    attrs = cmds.listAttr(yeti_shape_node,
+                          userDefined=True,
+                          string=("yetiVariableV_*",
+                                  "yetiVariableF_*")) or []
+    valid_attrs = []
+    for attr in attrs:
+        attr_type = cmds.attributeQuery(attr, node=yeti_shape_node,
+                                        attributeType=True)
+        if attr.startswith("yetiVariableV_") and attr_type == "double3":
+            # vector
+            valid_attrs.append(attr)
+        elif attr.startswith("yetiVariableF_") and attr_type == "double":
+            valid_attrs.append(attr)
+
+    return valid_attrs
+
+
+def create_yeti_variable(yeti_shape_node: str,
+                         attr_name: str,
+                         value=None,
+                         force_value: bool = False) -> bool:
+    """Get user defined yeti user variables for a `pgYetiMaya` shape node.
+
+    Arguments:
+        yeti_shape_node (str): The `pgYetiMaya` shape node.
+        attr_name (str): The fully qualified yeti variable name, e.g.
+            "yetiVariableF_myfloat" or "yetiVariableV_myvector"
+        value (object): The value to set (must match the type of the attribute)
+            When value is None it will ignored and not be set.
+        force_value (bool): Whether to set the value if the attribute already
+            exists or not.
+
+    Returns:
+        bool: Whether the attribute value was set or not.
+
+    """
+    exists = cmds.attributeQuery(attr_name, node=yeti_shape_node, exists=True)
+    if not exists:
+        if attr_name.startswith("yetiVariableV_"):
+            _create_vector_yeti_user_variable(yeti_shape_node, attr_name)
+        if attr_name.startswith("yetiVariableF_"):
+            _create_float_yeti_user_variable(yeti_shape_node, attr_name)
+
+    if value is not None and (not exists or force_value):
+        plug = "{}.{}".format(yeti_shape_node, attr_name)
+        if (
+                isinstance(value, (list, tuple))
+                and attr_name.startswith("yetiVariableV_")
+        ):
+            cmds.setAttr(plug, *value, type="double3")
+        else:
+            cmds.setAttr(plug, value)
+
+        return True
+    return False
+
+
+def _create_vector_yeti_user_variable(yeti_shape_node: str, attr_name: str):
+    if not attr_name.startswith("yetiVariableV_"):
+        raise ValueError("Must start with yetiVariableV_")
```

Comment:
Maybe we can raise the `LoadError` which can be implemented from the loader plugin in ayon pipeline?

Reply:
We could - but honestly, this should never actually occur during load because the extraction and load logic already ensures that basically these should be prefixed - specifically because they are called from `create_yeti_variable` which will only ever call one of the two if the names are valid.

The raising of the error is just there so that if someone were to somehow call these private functions and misuse them there would be warned about it.